components/sts/org.wso2.carbon.identity.sts.passive.ui/src/main/java/org/wso2/carbon/identity/sts/passive/ui/PassiveSTS.java

```diff
@@ -178,14 +179,14 @@ private void sendData(HttpServletResponse httpResp, ResponseToken respToken, Str
         String htmlPage = stsRedirectPage;
         String pageWithReply = htmlPage.replace("$url", String.valueOf(respToken.getReplyTo()));
 
-        String pageWithReplyAction = pageWithReply.replace("$action", String.valueOf(action));
+        String pageWithReplyAction = pageWithReply.replace("$action", Encode.forHtml(String.valueOf(action)));
         String pageWithReplyActionResult = pageWithReplyAction.replace("$result",
-                StringEscapeUtils.escapeHtml(String.valueOf(respToken.getResults())));
+                Encode.forHtml(String.valueOf(respToken.getResults())));
         String pageWithReplyActionResultContext;
         if(respToken.getContext() !=null) {
             pageWithReplyActionResultContext = pageWithReplyActionResult.replace(
                     "<!--$additionalParams-->", "<!--$additionalParams-->" + "<input type='hidden' name='wctx' value='"
```

Comment:
Here we replace the STS parameter values read by the sts_response.html page (which facilitates us to customize the Default STS Response page). Hence we are not explicitly generating any XML tags in the code.